index.ts

```diff
@@ -1,11 +1,19 @@
+import { readFileSync } from 'fs'
+import { jams } from 'jams.js'
 import { PackBuilder } from './src/builder'
 import { Dapp } from './src/dapp'
 import { getIpfsJson, putIpfsJson, isCid } from './src/ipfs-util'
 import { need } from './src/util'
 
 export const load = async (arg, _ethers = undefined, _signer = undefined) => {
   if (typeof arg === 'string') {
-    arg = (isCid(arg)) ? await getIpfsJson(arg) : require(arg)
+    if (isCid(arg)) {
+      arg = getIpfsJson(arg)
+    } else if (arg.split('.').pop() === 'jams') {
```

Comment:
This 3-way branch is not right, we have 2x2 options,

- Are we using a CID and fetching it from IPFS, or opening a local file?
- Is it JAMS or JSON?

Reply:
Does this work?

```typescript
if (isCid(arg)) {
  arg = getIpfsJson(arg)
} else {
 arg = (arg.split('.').pop() === 'jams')
    ? jams(readFileSync(arg))
    : readFileSync(arg)
}
```

Reply:
This one is missing the case that it is a CID that points to a JAMS file, maybe refactor `getIpfsJson` to just `getIpfsFile` then branch from there, point is there are 4 cases